### PR TITLE
Docs update

### LINKED ISSUE #259: Documentation inconsistencies / small fixes
Here is a list where I am keeping documentation inconsistencies / small fixes in Tangelo. Before the next release, we could fix them all at once.

- The Pauli strings are reversed in (code is correct) https://github.com/goodchemistryco/Tangelo/blob/f4d46ad0a5bb6c10b456e1d24dc8d8c8546837e7/tangelo/toolboxes/qubit_mappings/hcb.py#L67-L69
- lauren -> laurent
https://github.com/goodchemistryco/Tangelo/blob/5d3e9da841550dd30ebefd6c7f7d6abba5785930/tangelo/toolboxes/circuits/qsp.py#L49
- It is not a pyscf molecule.
https://github.com/goodchemistryco/Tangelo/blob/3559208427fb2eb55c5be53b1480f10b8f7008ca/tangelo/algorithms/variational/adapt_vqe_solver.py#L138

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
- Updated conf.py to 2023.
- Fixed #259.
- Removed gremlin characters.
- Added contributions and tutorials pages to the index for better reachability.
- Get version from the `_version.py` file.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -18,11 +18,11 @@
 # -- Project information -----------------------------------------------------
 
 project = 'tangelo'
-copyright = '2022 Good Chemistry Company'
+copyright = '2023 Good Chemistry Company'
 author = 'Valentin senicourt, Alexandre Fleury, Ryan Day, James Brown'
 
 # The full version, including alpha/beta/rc tags
-release = '0.3.1'
+release = '0.3.4'
```
Comment (ValentinS4t1qbit) [inline docs/source/conf.py:25]: Oh: think you can grab the  version string `_version` from inside the `tangelo` package ?
No worries if not, this is really a detail.

**Comment by ValentinS4t1qbit** [comment]
I don't know what precisely the goal of this PR is, but if there's no goal then we can merge this code as soon as we want. I'd rather do that. Nothing here is critical.

File: docs/source/conf.py (hunk 1 above)
Comment (AlexandreF-1qbit) [reply docs/source/conf.py:25]: I can try something!

**Comment by AlexandreF-1qbit** [comment]
@ValentinS4t1qbit there is no particular goal. It is an occasion to fix typos, links and other things in the docs after the release.

File: docs/source/index.rst [hunk 2]
```diff
@@ -11,6 +11,8 @@ Welcome to Tangelo's documentation!
    :caption: Contents:
 
    overview
+   TUTORIALS
```
Comment (ValentinS4t1qbit) [inline docs/source/index.rst:14]: Do these appear in all caps in our docs?
If yes I think we can simply solve this by renaming them with no caps.
Also, couldn't `docs/source/CONTRIBUTIONS.rst` (resp `TUTORIALS`) simply be symbolic links ?

**Review by ValentinS4t1qbit — COMMENTED** [review]
I think users should be able to easily find our example website / repo from this doc.
( https://goodchemistryco.github.io/Tangelo-Examples/ ). If it's not the case yet, we should probably reinforce that statement and use it again, in particular as a first sentence or so to "tutorials.rst".

File: docs/source/index.rst (hunk 2 above)
Comment (AlexandreF-1qbit) [reply docs/source/index.rst:14]: - No it doesn't. It is displaying the content of the file (thus no caps).
- CONTRIBUTIONS.rst and TUTUTIALS.rst are already symbolic links in `docs/source`, if I set it up correctly.

**Review by ValentinS4t1qbit — APPROVED** [review]
Thank you !

**Pull request merged**